blocks/api/factory.js

```diff
@@ -52,7 +54,7 @@ export function createBlock( name, blockAttributes = {} ) {
 		uid: uuid(),
 		name,
 		isValid: true,
-		attributes,
+		attributes: innerHTML ? getBlockAttributes( blockType, innerHTML, attributes ) : attributes,
```

Comment:
If it helps the discussion, `getBlockAttributes` is already part of the public interface of `@wordpress/blocks`; in other words, this change is more about the convenience we offer block authors with this polymorphic version of `createBlock` rather than keeping some piece of the blocks module private.